velox/exec/Task.h

```diff
@@ -640,6 +640,31 @@ class Task : public std::enable_shared_from_this<Task> {
     return terminateRequested_;
   }
 
+  /// Requests barrier on task execution to drain out all the buffered data
+  /// after finishes processing all the queued source splits. The returned
+  /// future will be realized when the drain operations finish. This is only
+  /// supported on task under single threaded execution mode and all its plan
+  /// nodes support barrier processing.
+  ContinueFuture requestBarrier();
+
+  /// Returns true if this task is under barrier processing.
+  bool underBarrier() const {
+    return barrierRequested_;
+  }
+
+  /// Sets to suppress the input for the specified operator. 'planNodeId' is the
+  /// associated plan node id. The function is called when a draining operator
+  /// needs no more input to finish the drain operation. All the upstream
+  /// operators can simply drop their output and output processing to finish the
+  /// drain quickly.
+  void suppressInput(const core::PlanNodeId& planNodeId, size_t operatorId);
```

Comment:
I believe this API is used by MergeJoin to suppress input from either left or right side. To suppress input from left side it calls this API with its own operator ID. To suppress input from the right side it calls this API with hard-coded operatorId 0. This feels a bit hacky. I wonder if it would be cleaner to model this API to align with existing create/getMergeJoinSource APIs? Perhaps, we could have merge-join-specific API that tells the Task that it doesn't need any further input to reach the barrier. The Task itself can be sort out which side of the join (if any) needs to drop further processing. What do you think?